checker/src/main/java/org/checkerframework/checker/calledmethods/EnsuredCalledMethodOnException.java

```diff
@@ -0,0 +1,65 @@
+package org.checkerframework.checker.calledmethods;
+
+import java.util.Objects;
+
+/**
+ * A contract that a method calls the given method on the given expression when that method throws
+ * an exception.
+ *
+ * <p>Instances of this class are plain old immutable data with no interesting behavior.
+ *
+ * @see org.checkerframework.checker.calledmethods.qual.EnsuresCalledMethodsOnException
+ */
+// TODO: In the future, this class should be a record.
+public class EnsuredCalledMethodOnException {
```

Comment:
To me it's slightly confusing that the name of this class differs by one character from the corresponding annotation.  Maybe `EnsuresCalledMethodOnExceptionContract`?  Not sure what is best here.